pkg/controller/controller.go

```diff
@@ -512,18 +579,37 @@ func (c *Controller) syncSharedConfigMap(event client.Event) error {
 		return fmt.Errorf("unexpected object vs. shared configmap: %v", event.Object.GetObjectKind().GroupVersionKind())
 	}
 	klog.V(4).Infof("verb %s share name %s", event.Verb, share.Name)
+	var err error
 	switch event.Verb {
 	case client.DeleteObjectAction:
-		objcache.DelSharedConfigMap(share)
+		ns := objcache.DelSharedConfigMap(share)
+		if c.refreshResources {
+			c.PruneConfigMapInformers(ns)
+
+		}
```

Comment:
This comment applies to all of the functions that you updated to return the namespaces.

I think you would be better off just creating a method that can be called from here to get the list of namespaces, instead of returning them from each function.

```suggestion
		objcache.DelSharedConfigMap(share)
		if c.refreshResources {
			c.PruneConfigMapInformers(NamespacesWithSharedConfigMaps())

		}
```